index.html

```diff
@@ -1246,334 +1093,763 @@ <h3>DID URL Content Metadata</h3>
+				</ul>
+			</li>
+
+			<li>
+				<strong>Handle Fragments:</strong>
+				<ul>
+					<li>If the <a>DID URL</a> contains a fragment, DO NOT
+					process or interpret it.</li>
+					<li>A <a>DID resolver</a> does not process the fragment as part
+					of its dereferencing behavior. Interpretation is the client's
+					responsibility based on the resolved document and media
+					type.</li>
+				</ul>
+			</li>
+
+			<li>
+				<strong>Resolve the DID:</strong>
+				<ul>
+					<li>Identify and apply query parameters from <code>qp</code>
+					that affect DID Resolution (e.g., <code>versionId</code>,
+					<code>versionTime</code>), including <a>DID Method</a>-specific
+					and <a>DID Extension</a> query parameters understood by the
+					<a>DID Resolver</a>.</li>
+					<li>Use these to resolve the <code>did</code> into a
+					<code>didDocument</code> and
+					<code>didResolutionMetadata</code>.</li>
+					<li>If resolution fails, return an error and abort
+					dereferencing.</li>
+					<li>Remove any consumed query parameters from the
+					<code>qp</code> working set.</li>
+				</ul>
+			</li>
+
+			<li>
+				<strong>No Path Component:</strong>
+				<ul>
+					<li>If no <code>path</code> is present, resolution is
+					complete and the path-handling step is skipped.</li>
+				</ul>
+			</li>
+
+			<li>
+				<strong>With Path Component:</strong>
+				<ul>
+					<li>Construct a target URL using the path resolution rules from
+					<a href="#did-url-path-processing">Section 5.4.3</a>.</li>
+					<li>Remove any consumed query parameters from the
+					<code>qp</code> working set.</li>
+					<li>If there are unused query parameters in <code>qp</code>, add
+					them to the constructed URL as query parameters, ensuring they
+					are formatted according to <a
+					href="https://datatracker.ietf.org/doc/html/rfc3986#section-3.4">RFC3986 Section 3.4</a>..</li>
+					<li>Attempt to resolve the resulting URL to retrieve the
+					resource and metadata to be returned to the client.</li>
+					<li>If resolution fails, return an error and abort
+					dereferencing.</li>
+				</ul>
+				<ul>
+					<li>If the resolution of the URL fails, an error
+					MUST be returned.</li>
+					<li>If the <a>DID Resolver</a> is unable to resolve the
+					URL, an error MUST be returned.</li>
+				</ul>
+			</li>
 
+			<li>
+				<strong>Return the dereferencing result:</strong>
+				<ul>
+				<li>
+					If the resolution result is a <a>DID Document</a>,
+					return:
+					<ul>
+						<li><code>content</code>: the resolved DID document
+						with selected services</li>
+						<li><code>contentMetadata</code>: metadata about the
+						resolved DID document</li>
+						<li><code>dereferencingMetadata</code>: any metadata
+						about the dereferencing process</li>
+					</ul>
+				</li>
+				<li>
+					If the resolution result is a resource resolved via DID URL path processing, return:
+					<ul>
+						<li><code>contentStream</code>: the resolved
```

Comment:
Are you sure we want to inline the content stream into the resulting json object? What if it's like a large video or image?